source/aws_imds_client.c

```diff
@@ -509,7 +513,8 @@ static void s_client_on_token_response(struct imds_user_data *user_data) {
      * we should fall back to insecure request. Otherwise, we should use
      * token in following requests.
      */
-    if (user_data->status_code != AWS_HTTP_STATUS_CODE_200_OK || user_data->current_result.len == 0) {
+    if (!user_data->ec2_metadata_v1_disabled &&
```

Comment:
I would prefer to be more clear about this:
```suggestion
    if (user_data->status_code != AWS_HTTP_STATUS_CODE_200_OK || user_data->current_result.len == 0) {
    if (user_data->ec2_metadata_v1_disabled) {
       //XXX failed to fetch creds and fallback disabled
    }
    else {
       //XXX Fallback to retry
    }
}
```

Reply:
I think creating another branch of code makes it more confusing to follow. I have improved the documentation.

Reply:
Thanks, I have refactored the code to the following.
```
	if(success){
	} else if(fallback disabled){
		// error out
	} else {
		fallback to v1.
	}
```